### PR TITLE
Add docs for building the plugin

### PR DESCRIPTION
**Pull request by swissspidy**
**Note:** this PR is best reviewed by looking at the individual commits.

Here's what it does:

* [Add docs for building the plugin](https://github.com/WordPress/plugin-check/commit/9b9aedcd92083521aa0b8fff66e732a6cc8656b2)
* [Change version to 1.0.0 everywhere](https://github.com/WordPress/plugin-check/commit/2615834d52bd4b9cf78f100f426291a423f6819c)
	I don't think we need this whole `n.e.x.t` thing with an extra build step just to bump version numbers.
* [Add .distignore file for building the plugin](https://github.com/WordPress/plugin-check/commit/4e6ca435fcd63dcc84225cf8225aa50dc1a2b16f)
* [Ensure dependencies are installed in deploy step](https://github.com/WordPress/plugin-check/commit/2052378a780f34f4143832b1409f4345934dce99)
* [Remove dates from changelog](https://github.com/WordPress/plugin-check/commit/d5a2fd07a333d7bed0ab5dd0a6b74548dbcbba5c)
	* Causes less pressure for shipping on time :)
* [Expand the docs to cover the releasing part](https://github.com/WordPress/plugin-check/commit/797453e172f0f33adfb883777ea5656e859276dd)

Here's a manually created ZIP file for testing:

[plugin-check.zip](https://github.com/WordPress/plugin-check/files/14184375/plugin-check.zip)

### CODE REVIEW

**Comment by swissspidy** [comment]
@mukeshpanchal27 What's up with this failing `Plugin_Readme_Check_Tests::test_run_with_errors_empty_name` test on trunk? 🤔

File: cli.php [hunk 1]
```diff
@@ -44,10 +44,10 @@
  */
 WP_CLI::add_hook(
 	'before_wp_load',
-	function() {
+	function () {
 		if ( CLI_Runner::is_plugin_check() ) {
 			if ( ! file_exists( ABSPATH . 'wp-content/object-cache.php' ) ) {
-				if ( ! copy(  __DIR__ . '/drop-ins/object-cache.copy.php', ABSPATH . 'wp-content/object-cache.php' ) ) {
+				if ( ! copy( __DIR__ . '/drop-ins/object-cache.copy.php', ABSPATH . 'wp-content/object-cache.php' ) ) {
```
Comment (westonruter) [inline cli.php:50]: Curious why these `if` statements aren't all combined into one?

**Review by westonruter — APPROVED** [review]
LGTM

**Comment by ernilambar** [comment]
> @mukeshpanchal27 What's up with this failing `Plugin_Readme_Check_Tests::test_run_with_errors_empty_name` test on trunk? 🤔

See my comment https://github.com/WordPress/plugin-check/pull/417

File: .distignore [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+build
+build-cs
+build-phpunit
+docs
+node_modules
+test-content
+tests
+
+*.DS_store
+.DS_store
```
Comment (ernilambar) [inline .distignore]: .DS_Store is still there in the zip.

File: .distignore [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+.github
+.idea
+.wordpress-org
+build
+build-cs
+build-phpunit
+docs
+node_modules
+test-content
+tests
```
Comment (ernilambar) [inline .distignore:12]: `patches` folder should also be excluded I believe.

**Review by bordoni — APPROVED** [review]

**Pull request merged**